pkg/container-utils/containerutils.go

```diff
@@ -117,3 +120,67 @@ func ParseOCIState(stateBuf []byte) (id string, pid int, err error) {
 	pid = ociState.Pid
 	return
 }
+
+// GetIfacePeers returns the networking interfaces on the host side of the container where pid is
+// running in.
+func GetIfacePeers(pid int) ([]*net.Interface, error) {
+	var ifaceLinks []int
+
+	err := netnsenter.NetnsEnter(pid, func() error {
+		interfaces, err := net.Interfaces()
+		if err != nil {
+			return fmt.Errorf("getting interfaces: %w", err)
+		}
+
+		for _, iface := range interfaces {
+			// Ignore loopback and down interfaces
+			if iface.Flags&net.FlagLoopback != 0 || iface.Flags&net.FlagUp == 0 {
+				continue
+			}
+
+			path := fmt.Sprintf("/proc/%d/cwd/sys/class/net/%s/iflink", pid, iface.Name)
```

Comment:
* Need `HostProcFs` instead of `/proc`
* Why `cwd` and not `root`?
* Does this work on containers with a bind mount on /sys? cadvisor uses `docker run --volume=/sys:/sys:ro ...` (see [docs/running.md#L12](https://github.com/google/cadvisor/blob/42bb3d13a0cf9ab80c880a16c4ebb4f36e51b0c9/docs/running.md#L12)). I think you have to use [VethPeerIndex](https://github.com/vishvananda/netlink/blob/06219cde3e811f757f263ae54e848891c8ce6cbc/link_linux.go#L3664C6-L3664C19) instead.

Reply:
> Need HostProcFs instead of /proc

Line below prepends `HostRoot`

> Why cwd and not root?

I suppose for systemd both work fine as it'll always be running on `/`? Do you suggest to use root instead?

> Does this work on containers with a bind mount on /sys? 

I tried by adding this to a pod where ig is run and seems to work fine:

```bash
$ mount | grep /sys | head
sysfs on /sys type sysfs (ro,relatime)
```

> I think you have to use [VethPeerIndex](https://github.com/vishvananda/netlink/blob/06219cde3e811f757f263ae54e848891c8ce6cbc/link_linux.go#L3664C6-L3664C19) instead.

Thanks for the suggestion, I was looking at it before but I was not sure about how to use it. Updated the code and it works fine. ,